File: packages/typescript/src/client/zsxq-client.ts
import { ZsxqConfig } from './zsxq-config';
import { HttpClient } from '../http';
import {
  GroupsRequest,
  TopicsRequest,
  UsersRequest,
  CheckinsRequest,
  DashboardRequest,
} from '../request';

/**
 * 知识星球 SDK 主客户端
 *
 * 使用 ZsxqClientBuilder 构建实例：
 * ```typescript
 * const client = new ZsxqClientBuilder()
 *   .setToken('your-token')
 *   .build();
 *
 * // 获取星球列表
 * const groups = await client.groups.list();
 *
 * // 获取话题
 * const topics = await client.topics.list(groupId);
 * ```
 */
export class ZsxqClient {
  /** 星球管理 */
  readonly groups: GroupsRequest;

  /** 话题管理 */
  readonly topics: TopicsRequest;

  /** 用户管理 */
  readonly users: UsersRequest;

  /** 打卡管理 */
  readonly checkins: CheckinsRequest;

  /** 数据面板 */
  readonly dashboard: DashboardRequest;

  private readonly httpClient: HttpClient;

  constructor(config: ZsxqConfig) {
    this.httpClient = new HttpClient(config);

    // 初始化各功能模块
    this.groups = new GroupsRequest(this.httpClient);
    this.topics = new TopicsRequest(this.httpClient);
    this.users = new UsersRequest(this.httpClient);
    this.checkins = new CheckinsRequest(this.httpClient);
    this.dashboard = new DashboardRequest(this.httpClient);
  }
}

File: packages/java/src/main/java/com/zsxq/sdk/client/ZsxqClient.java
package com.zsxq.sdk.client;

import com.zsxq.sdk.http.HttpClient;
import com.zsxq.sdk.request.*;

/**
 * 知识星球 SDK 主客户端
 *
 * 使用 ZsxqClientBuilder 构建实例：
 * <pre>
 * ZsxqClient client = new ZsxqClientBuilder()
 *     .token("your-token")
 *     .build();
 *
 * // 获取星球列表
 * List&lt;Group&gt; groups = client.groups().list();
 *
 * // 获取话题
 * List&lt;Topic&gt; topics = client.topics().list(groupId);
 * </pre>
 */
public class ZsxqClient {

    private final GroupsRequest groups;
    private final TopicsRequest topics;
    private final UsersRequest users;
    private final CheckinsRequest checkins;
    private final DashboardRequest dashboard;
    private final RankingRequest ranking;
    private final MiscRequest misc;

    ZsxqClient(ZsxqConfig config) {
        HttpClient httpClient = new HttpClient(config);

        this.groups = new GroupsRequest(httpClient);
        this.topics = new TopicsRequest(httpClient);
        this.users = new UsersRequest(httpClient);
        this.checkins = new CheckinsRequest(httpClient);
        this.dashboard = new DashboardRequest(httpClient);
        this.ranking = new RankingRequest(httpClient);
        this.misc = new MiscRequest(httpClient);
    }

    /**
     * 星球管理
     */
    public GroupsRequest groups() {
        return groups;
    }

    /**
     * 话题管理
     */
    public TopicsRequest topics() {
        return topics;
    }

    /**
     * 用户管理
     */
    public UsersRequest users() {
        return users;
    }

    /**
     * 打卡管理
     */
    public CheckinsRequest checkins() {
        return checkins;
    }

    /**
     * 数据面板
     */
    public DashboardRequest dashboard() {
        return dashboard;
    }

    /**
     * 排行榜
     */
    public RankingRequest ranking() {
        return ranking;
    }

    /**
     * 杂项功能
     */
    public MiscRequest misc() {
        return misc;
    }
}
